File: Stage10/Problem5073.py
def main():
    while True:
        side1, side2, side3 = map(int, input().split())

        if side1 == 0 and side2 == 0 and side3 == 0:
            break
        
        sides = [side1, side2, side3]
        max_side = max(sides)
        total = sum(sides)
        
        if total - max_side <= max_side:
            print("Invalid")
        elif side1 == side2 == side3:
            print("Equilateral")
        elif side1 == side2 or side2 == side3 or side1 == side3:
            print("Isosceles")
        else:
            print("Scalene")
            
    
if __name__ == "__main__":
    main()

File: Stage10/C/Problem5073.c
#include <stdio.h>

int main(void) {
    int side1, side2, side3;
    while (1) {
        scanf("%d %d %d", &side1, &side2, &side3);

        if (side1 == 0 && side2 == 0 && side3 == 0) break;

        int max = side1;
        if (side2 > max) max = side2;
        if (side3 > max) max = side3;
        int sum = side1 + side2 + side3;

        if (sum - max <= max) {
            printf("Invalid\n");
        } else if (side1 == side2 && side2 == side3) {
            printf("Equilateral\n");
        } else if (side1 == side2 || side2 == side3 || side1 == side3) {
            printf("Isosceles\n");
        } else {
            printf("Scalene\n");
        }
    }

    return 0;
}